sys/usb/Kconfig

```diff
@@ -42,20 +53,35 @@ config USB_SPEC_BCDVERSION_1_1
 
 endchoice
 
+config USB_USED_IN_APPLICATION
```

Comment:
I still don`t understand the purpose of this option. Should it be understood like `USB_USE_CUSTOM_VID_PID`? The help text does not really help to clarify this. What exactly means `enable the USB peripheral in the application`? An application that enables `KCONFIG_USB` and selects an implementation and according device classes already uses the USB peripheral.

Reply:
Maybe we don't need it if we just use the check in the .h file.  I will spend some time tomorrow cleaning it up and we can see.

The original purpose of this is to enforce the internal peripheral PID vs application PID check from the kconfig side...

Reply:
We do need this actually.  The logic we currently have is basically checking if anything is defined for `CONFIG_USB_PID` and `CONFIG_USB_VID`.  If we don't block those values in Kconfig they will be exported and the `usb.h` macro check will fail...

I have changed it to more of the convention that we use `CUSTOM_USB_VID_PID` and hopefully documented it better.

If one wants to use the `usb.h` in their application without specifying `CUSTOM_USB_VID_PID` it will throw the error saying they need to config it, which is what we want.  By default those values will be the `TESTING_USB*` values but we can change the default depending on the on the riot boot module.